src/scheduler/server_info.c

```diff
@@ -223,8 +223,12 @@ query_server(status *pol, int pbs_sd)
 		return NULL;
 	}
 
+	sched = pbs_statsched(pbs_sd, NULL, NULL);
+	while(sched && strncmp(PBS_DFLT_SCHED_NAME, sched->name, PBS_MAXSCHEDNAME)) {
+		sched = sched->next;
+	}
 
-	if ((sched = pbs_statsched(pbs_sd, PBS_DFLT_SCHED_NAME, NULL, NULL)) == NULL) {
+	if (sched == NULL) {
```

Comment:
Move this check up above the loop

Reply:
I think this check is at the right place. after looping the returned shcedstat batch_status linked list, we may end-up with `NULL` in `sched` if none of the sched stat match with sc_name